src/Filters/Issue.jsx

```diff
@@ -32,11 +47,65 @@ export class Issue extends React.Component {
       </CollapsibleFilter>
     )
   }
+
+  // --------------------------------------------------------------------------
+  // Typeahead Helpers
+
+  _onInputChange(value) {
+    // Normalize the input value 
+    const normalized = normalize(value)
+
+    // Find the matches
+    const filtered = this.props.forTypeahead
+      .filter(x => x.normalized.indexOf(normalized) !== -1)
+      .map(x => {
+        return {
+          key: x.key,
+          position: x.normalized.indexOf(normalized),
+          value
+        }
+      })
+
+    // Sort the matches so that matches at the beginning of the string
+    // appear first
+    filtered.sort((a,b) => {
+      return a.position - b.position
+    })
+
+    return filtered
+  }
+
+  _renderOption(obj) {
+    const start = obj.key.substring(0, obj.position)
+    const match = obj.key.substr(obj.position, obj.value.length)
+    const end = obj.key.substring(obj.position + obj.value.length)
+
+    return {
+      value: obj.key,
+      component: (
+        <span>{start}<b>{match}</b>{end}</span>
+      )
+    }
+  }
+
+  _onOptionSelected(obj) {
+    // TODO: Select the parent + children
+    console.log('Selected "', obj.key, '"')
+  }
 }
 
 export const mapStateToProps = state => {
+  const options = state.aggs.issue || []
+  const forTypeahead = options.map(x => {
+    return {
+      key: x.key,
+      normalized: normalize(x.key)
+    }
+  })
+
   return {
-    options: state.aggs.issue
+    options,
+    forTypeahead
   }
```

Comment:
I knew we had to refactor the parent/child selection logic, so I figured I would handle that TODO when I update the logic